mmhuman3d/core/visualization/renderer/torch3d_renderer/meshes.py

```diff
@@ -0,0 +1,540 @@
+            self.face_individual = faces[:1, :self.model_class.NUM_FACES].to(
+                device)
+        else:
+            raise ValueError(f'Wrong type of faces: {type(faces)}.')
+
+        assert faces.shape == (N,
+                               self.model_class.NUM_FACES * self._N_individual,
+                               3)
+        F = faces.shape[1]
+        if textures is None:
+            if texture_images is None:
+                # input vertex_color should be
+                #   (3), (1, 3), (1, 1, 3). all the same color
+                #   (V, 3), (1, V, 3), each vertex has a single color
+                #   (N, V, 3), each batch each vertex has a single color
+                if isinstance(vertex_color, (tuple, list)):
+                    vertex_color = torch.Tensor(vertex_color)
+                elif isinstance(vertex_color, (torch.Tensor, np.ndarray)):
+                    vertex_color = torch.Tensor(vertex_color) if isinstance(
+                        vertex_color, np.ndarray) else vertex_color
+                if vertex_color.numel() == 3:
+                    vertex_color = vertex_color.view(1, 3).repeat(V, 1)
+                vertex_color = align_input_to_padded(
+                    vertex_color, ndim=3, batch_size=N)
+                assert vertex_color.shape == verts.shape
+                if use_nearest:
+                    textures = TexturesNearest(
+                        verts_features=vertex_color).to(device)
+                else:
+                    textures = TexturesVertex(
+                        verts_features=vertex_color).to(device)
+            else:
+
+                texture_images = align_input_to_padded(
+                    texture_images, ndim=4, batch_size=N).to(device)
+
+                assert uv_renderer is not None
+                if isinstance(uv_renderer, dict):
+                    uv_renderer = build_renderer(uv_renderer)
+                uv_renderer = uv_renderer.to(device)
+                textures = uv_renderer.wrap_texture(texture_images).to(device)
+                if self._N_individual > 1:
+                    textures = textures.join_scene()
+                    textures = textures.extend(N)
+
+        num_verts_per_mesh = [V for _ in range(N)]
+        num_faces_per_mesh = [F for _ in range(N)]
+
+        if use_list:
+            verts = padded_to_list(verts, num_verts_per_mesh)
+            faces = padded_to_list(faces, num_faces_per_mesh)
+        super().__init__(
+            verts=verts,
+            faces=faces,
+            textures=textures,
+            verts_normals=verts_normals,
+        )
+
+    def get_faces_padded(self, N_batch, N_individual):
+        faces = self.face_individual.repeat(N_batch, N_individual, 1)
+        faces_offset = torch.arange(N_individual).view(N_individual, 1).repeat(
+            1, self.model_class.NUM_FACES).view(1, -1, 1).to(faces.device)
+        faces += faces_offset
```

Comment:
```suggestion
        faces = faces + faces_offset * self.model_class.NUM_VERTS
```
Values in `faces_offset` are [0,...., 0, 1,....., 1....], thus we need NUM_VERTS to get correct mapping.

Reply:
Thanks for finding this bug